stream_alert/alert_processor/helpers.py

```diff
@@ -30,3 +31,81 @@ def elide_string_middle(text, max_length):
 
     half_len = (max_length - 5) / 2  # Length of text on either side.
     return '{} ... {}'.format(text[:half_len], text[-half_len:])
+
+
+def publish_alert(alert, output, descriptor):
+    """Presents the alert as a dict for output classes to send to their API integrations.
+
+    Args:
+        alert (Alert): The alert to be dispatched
+        output (OutputDispatcher): Instance of the output class dispatching this alert
+        descriptor (str): The descriptor for the output
+
+    Returns:
+        dict
+    """
+    publisher = _assemble_alert_publisher_for_output(
+        alert,
+        output,
+        descriptor
+    )
+    return publisher.publish(alert, {})
+
+
+def _assemble_alert_publisher_for_output(alert, output, descriptor):
+    """Gathers all requested publishers on the alert and returns them as a single Publisher
+
+    Args:
+        alert (Alert): The alert that is pulled from DynamoDB
+        output (OutputDispatcher|None): Instance of OutputDispatcher that is sending the alert
+        descriptor (str): The descriptor of the Output
+
+    Returns:
+        AlertPublisher
+    """
+
+    alert_publishers = alert.publishers
+    publisher_names = []
+    if isinstance(alert_publishers, basestring):
+        # Case 1: The publisher is a single string.
+        #   apply this single publisher to all outputs + descriptors
+        publisher_names.append(alert_publishers)
```

Comment:
is this still accurate? string literal publisher names and not classes/functions?

Reply:
In this specific file, this asks about `alert.publishers` which is a field on DynamoDb, so they are always string literals and never classes/functions. They have to be looked-up and transformed into the actual `Publisher` classes using the `create_composite_publisher`